From: Patrick Brunschwig <patrick@enigmail.net>
Date: Thu, 29 Mar 2018 17:26:14 +0200
Subject: ensure lowercase for all email addresses

(cherry picked from commit d2caf2b0fed5714fe4fb86b26ee3376382fd6d6a)
---
 package/autocrypt.jsm | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/package/autocrypt.jsm b/package/autocrypt.jsm
index 726603e..7b86c83 100644
--- a/package/autocrypt.jsm
+++ b/package/autocrypt.jsm
@@ -108,7 +108,9 @@ var EnigmailAutocrypt = {
           }
         }
 
-        if (fromAddr !== paramArr.addr.toLowerCase()) {
+        paramArr.addr = paramArr.addr.toLowerCase();
+
+        if (fromAddr !== paramArr.addr) {
           EnigmailLog.DEBUG("autocrypt.jsm: processAutocryptHeader: from Addr " + fromAddr + " != " + paramArr.addr.toLowerCase() + "\n");
 
           resolve(3);
@@ -712,7 +714,7 @@ function appendUser(connection, paramsArr) {
   connection.executeTransaction(function _trx() {
     connection.execute("insert into autocrypt_keydata (email, keydata, fpr, type, last_seen_autocrypt, last_seen, state) values " +
       "(:email, :keyData, :fpr, :type, :lastAutocrypt, :lastSeen, :state)", {
-        email: paramsArr.addr,
+        email: paramsArr.addr.toLowerCase(),
         keyData: paramsArr.keydata,
         fpr: ("fpr" in paramsArr ? paramsArr.fpr : ""),
         type: paramsArr.type,
@@ -779,7 +781,7 @@ function updateUser(connection, paramsArr, resultRows, autoCryptEnabled) {
     updateStr = "update autocrypt_keydata set state = :state, keydata = :keyData, last_seen_autocrypt = :lastAutocrypt, " +
       "fpr = :fpr, last_seen = :lastSeen where email = :email and type = :type";
     updateObj = {
-      email: paramsArr.addr,
+      email: paramsArr.addr.toLowerCase(),
       state: paramsArr["prefer-encrypt"],
       keyData: paramsArr.keydata,
       fpr: ("fpr" in paramsArr ? paramsArr.fpr : ""),
@@ -791,7 +793,7 @@ function updateUser(connection, paramsArr, resultRows, autoCryptEnabled) {
   else {
     updateStr = "update autocrypt_keydata set state = :state, last_seen = :lastSeen where email = :email and type = :type";
     updateObj = {
-      email: paramsArr.addr,
+      email: paramsArr.addr.toLowerCase(),
       state: paramsArr["prefer-encrypt"],
       type: paramsArr.type,
       lastSeen: paramsArr.dateSent.toJSON()
